test/e2e/storage/volumes.go

```diff
@@ -59,6 +59,65 @@ import (
 	vspheretest "k8s.io/kubernetes/test/e2e/storage/vsphere"
 )
 
+type volumeTest struct {
+	name    string
+	config  framework.VolumeTestConfig
+	fsGroup *int64
+	tests   []framework.VolumeTest
+}
+
+var expectedContents = map[string]string{
+	"nfs":             "Hello from NFS!",       // content of test/images/volumes-tester/nfs/index.html
+	"gluster":         "Hello from GlusterFS!", // content of test/images/volumes-tester/gluster/index.html
+	"iscsi":           "Hello from iSCSI",      // content of test/images/volumes-tester/iscsi/block.tar.gz
+	"rdb":             "Hello from RBD",        // content of test/images/volumes-tester/rbd/create_block.sh
+	"ceph":            "Hello Ceph!",           // content of test/images/volumes-tester/rbd/create_block.sh
+	"hostpath":        "Hello hostpath!",
+	"hostpathSymlink": "Hello hostpathSymlink!",
+	// emtpydir doesn't provide persistency, therefore it should not be tested.
+}
+
+var needInjectHTML = map[string]bool{
```

Comment:
This should be part of `driverInfo`.

Reply:
I think that needInjectHTML is volumes.go's test specific configuration because whether injectHTML should be called in the tests is up to the tests themselves, therefore I prefer to keep it in test side, not driver side. I'm thinking about refactoring a bit more on each tests, so let's discuss it more after I refactor it.